src/plugins/intel_cpu/src/nodes/fullyconnected.cpp

```diff
@@ -329,6 +330,172 @@ void FullyConnected::prepackMLASWeight() {
+    }
+    auto outDesc = key.out->getDnnlDesc();
+    const auto& outDims = outDesc.get_dims(); // @TODO query + copy might be slow
+
+    if (outDims.size() == 3) {
+        auto normalizedOutDims = { outDims[0] * outDims[1], outDims[2] };
+        outDesc = outDesc.reshape(normalizedOutDims);
+    }
+    auto wghDescAny = dnnl::memory::desc(DnnlExtensionUtils::convertToDnnlDims(key.inp1->getShape().getStaticDims()),
+                                         key.inp1->getDataType(), memory::format_tag::any);
+    dnnl::inner_product_forward::primitive_desc prim_desc;
+    if (key.bias) {
+        prim_desc = dnnl::inner_product_forward::primitive_desc(
+            engine,
+            dnnl::prop_kind::forward_inference,
+            inDesc,
+            wghDescAny,
+            key.bias->getDnnlDesc(),
+            outDesc,
+            key.attr);
+    } else {
+        prim_desc = dnnl::inner_product_forward::primitive_desc(
+            engine,
+            dnnl::prop_kind::forward_inference,
+            inDesc,
+            wghDescAny,
+            outDesc,
+            key.attr);
+    }
+    auto first_desc = dnnl::inner_product_forward::primitive_desc(prim_desc.get());
+    const bool found = DnnlExtensionUtils::find_implementation(prim_desc, key.implType);
+
+    if (found)
+        return std::move(prim_desc);
+
+    return std::move(first_desc);
+}
+
+#if defined(OV_CPU_WITH_ACL)
+/**
+ * Do not wait till prepareParams to reorder the weights
+ * Do it in scope of compile_model using dummy shapes
+ */
+void FullyConnected::prepareWeightsUsingDummyShape() {
+    auto srcMemPtr = getParentEdgesAtPort(0)[0]->getMemoryPtr();
+    auto dstMemPtr = getChildEdgesAtPort(0)[0]->getMemoryPtr();
+    if (!dstMemPtr || !dstMemPtr->isAllocated())
+        IE_THROW() << "Destination memory hasn't been allocated.";
+    if (!srcMemPtr || !srcMemPtr->isAllocated())
+        IE_THROW() << "Input memory hasn't been allocated.";
+    MemoryPtr biasMemPtr = nullptr;
+    if (withBiases) {
+        biasMemPtr = getParentEdgesAtPort(2)[0]->getMemoryPtr();
+        if (!biasMemPtr || !biasMemPtr->isAllocated())
+            IE_THROW() << "Input memory hasn't been allocated.";
+    }
+    NodeDesc *selected_pd = getSelectedPrimitiveDescriptor();
+    if (selected_pd == nullptr)
+        IE_THROW() << "Preferable primitive descriptor is not set for node " << getName() << ".";
+
+    DnnlMemoryDescPtr weightDesc = MemoryDescUtils::convertToDnnlMemoryDesc(weightDescIP);
+    DnnlMemoryDescCPtr biasDesc = nullptr;
+    if (biasMemPtr) {
+        biasDesc = biasMemPtr->getDescWithType<DnnlMemoryDesc>();
+    }
+    DnnlMemoryDescCPtr inDesc = MemoryDescUtils::convertToDnnlMemoryDesc(
+        MemoryDescUtils::makeDummyDesc(*srcMemPtr->getDescWithType<DnnlMemoryDesc>()));
+    DnnlMemoryDescCPtr outDesc = MemoryDescUtils::convertToDnnlMemoryDesc(
+        MemoryDescUtils::makeDummyDesc(*dstMemPtr->getDescWithType<DnnlMemoryDesc>()));
```

Comment:
I would say it is more reliable to use memory descriptors from the port descriptors in `selected_pd `. The idea is that these memory descriptors are always undefined in the case of dynamic shapes and used specially for this purpose: creating defined memory descriptors from undefined ones with a specific shape. Please refer to: https://github.com/openvinotoolkit/openvino/blob/eacdc24d541eb800284b8d49d6fb1fae37cdce0b/src/plugins/intel_cpu/src/node.cpp#L634

Reply:
actually this was the first thing I have tried.
But onednn primitive_desc failed to create using memory descs from the selected pd

Reply:
Updated as discussed (better to use selected_pd).